src/goto-instrument/contracts/contracts.cpp

```diff
@@ -395,22 +395,20 @@ void code_contractst::replace_old_parameter(
     const auto &parameter = to_old_expr(expr).expression();
 
     // TODO: generalize below
-    if(parameter.id() == ID_dereference)
+    if(parameter.id() == ID_dereference || parameter.id() == ID_member)
```

Comment:
I think in the assigns clause code I also saw `ID_ptrmember`. I thought that `a.b` has `ID_member` and `a->b` has `ID_ptrmember`, but the regression tests you added seem to work with just `ID_member` check, so I might be wrong.
But do we need an `ID_ptrmember` case here?

Reply:
I added more tests to cover these cases. `ID_member` seems enough.

Reply:
I did a quick search, and it looks like `ID_ptrmember` indeed corresponds to `->`:
https://github.com/diffblue/cbmc/blob/1ab5de1ac0893c7ca94ffe5463e2c7c4ee781266/src/cpp/cpp_typecheck_expr.cpp#L454

Reply:
@SaswatPadhi This is for `cpp` not `ansi-c`.

Reply:
@feliperodri Are you sure? From the ANSI-C parser:

https://github.com/diffblue/cbmc/blob/5592d7e414ee3f440631f7281073345516391d75/src/ansi-c/parser.y#L565-L569